### PR TITLE
RIA-3473 mark appeal as paid for appeal types EA/HU/PA

### PR DESCRIPTION
**Pull request by tarun-palisetty**
### JIRA link (if applicable) ###
[RIA-3473](https://tools.hmcts.net/jira/browse/RIA-3473)

### Change description ###
Mark appeal as paid for the appeal types EA/HU/PA

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidStateHandler.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        final State currentState =
+            callback
+                .getCaseDetails()
+                .getState();
+
+        asylumCase.write(PAYMENT_STATUS, PaymentStatus.PAID);
+
+        String paymentDate = asylumCase.read(PAID_DATE, String.class)
+            .orElseThrow(() -> new IllegalStateException("Paid date is not present"));
+        asylumCase.write(PAYMENT_DATE,
```
Comment (Lee365) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidStateHandler.java:66]: Reading and writing PAID_DATE with format. Is this required if its written in the right format ?

**Review by Lee365 — COMMENTED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidHandler.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_START
+               && callback.getEvent() == Event.MARK_APPEAL_PAID;
```
[the remark this answers is not in the document]
Comment (tarun-palisetty) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidHandler.java:36]: isfeePaymentEnabled check added

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidHandler.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.PaymentStatus;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+@Component
+public class MarkPaymentPaidHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final boolean isfeePaymentEnabled;
```
[the remark this answers is not in the document]
Comment (tarun-palisetty) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidHandler.java:20]: isfeePaymentEnabled check added

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidHandler.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tarun-palisetty) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidHandler.java:36]: isfeePaymentEnabled check added

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidHandler.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (tarun-palisetty) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidHandler.java:20]: isfeePaymentEnabled check added

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidStateHandler.java (hunk 1 above)
Comment (tarun-palisetty) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidStateHandler.java:66]: When writing to different place holder the format is lost, so I'm formatted it before writing

**Review by Lee365 — APPROVED** [review]
LGTM

**Review by Lee365 — APPROVED** [review]
LGTM

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**